File: protect/src/main/java/com/webileapps/safeguard/CyberUtils.java
package com.webileapps.safeguard;

import android.content.Context;
import android.widget.Toast;
import java.util.function.Consumer;

public class CyberUtils {
    public static void showToast(Context context, String message) {
        Toast.makeText(context, message, Toast.LENGTH_SHORT).show();
    }

    public static void checkRoot(Context context, SecurityChecker securityChecker, Consumer<Boolean> onChecked) {
        SecurityChecker.SecurityCheck result = securityChecker.checkRootStatus();
        if (result instanceof SecurityChecker.SecurityCheck.Critical) {
            SecurityChecker.SecurityCheck.Critical critical = (SecurityChecker.SecurityCheck.Critical) result;
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(
                AppActivity.getContext(), 
                critical.message, 
                true,
                null
            );
            onChecked.accept(false);
        } else if (result instanceof SecurityChecker.SecurityCheck.Warning) {
            SecurityChecker.SecurityCheck.Warning warning = (SecurityChecker.SecurityCheck.Warning) result;
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(
                AppActivity.getContext(), 
                warning.message, 
                false,
                userAcknowledged -> {
                    if (userAcknowledged) {
                        onChecked.accept(true);
                    }
                }
            );
        } else {
            onChecked.accept(true);
        }
    }

    public static void checkDeveloperOptions(Context context, SecurityChecker securityChecker, Consumer<Boolean> onChecked) {
        SecurityChecker.SecurityCheck result = securityChecker.checkDeveloperOptions();
        if (result instanceof SecurityChecker.SecurityCheck.Critical) {
            SecurityChecker.SecurityCheck.Critical critical = (SecurityChecker.SecurityCheck.Critical) result;
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(
                AppActivity.getContext(), 
                critical.message, 
                true,
                null
            );
            onChecked.accept(false);
        } else if (result instanceof SecurityChecker.SecurityCheck.Warning) {
            SecurityChecker.SecurityCheck.Warning warning = (SecurityChecker.SecurityCheck.Warning) result;
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(
                AppActivity.getContext(), 
                warning.message, 
                false,
                userAcknowledged -> {
                    if (userAcknowledged) {
                        onChecked.accept(true);
                    }
                }
            );
        } else {
            onChecked.accept(true);
        }
    }

    public static void checkMalware(Context context, SecurityChecker securityChecker, Consumer<Boolean> onChecked) {
        SecurityChecker.SecurityCheck result = securityChecker.checkMalwareAndTampering();
        if (result instanceof SecurityChecker.SecurityCheck.Critical) {
            SecurityChecker.SecurityCheck.Critical critical = (SecurityChecker.SecurityCheck.Critical) result;
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(
                AppActivity.getContext(), 
                critical.message, 
                true,
                null
            );
        } else if (result instanceof SecurityChecker.SecurityCheck.Warning) {
            SecurityChecker.SecurityCheck.Warning warning = (SecurityChecker.SecurityCheck.Warning) result;
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(
                AppActivity.getContext(), 
                warning.message, 
                false,
                userAcknowledged -> {
                    if (userAcknowledged) {
                        onChecked.accept(true);
                    }
                }
            );
        } else {
            onChecked.accept(true);
        }
    }

    public static void checkScreenMirroring(Context context, SecurityChecker securityChecker, Consumer<Boolean> onChecked) {
        SecurityChecker.SecurityCheck result = securityChecker.checkScreenMirroring();
        if (result instanceof SecurityChecker.SecurityCheck.Warning) {
            SecurityChecker.SecurityCheck.Warning warning = (SecurityChecker.SecurityCheck.Warning) result;
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(
                AppActivity.getContext(), 
                warning.message, 
                false,
                userAcknowledged -> {
                    if (userAcknowledged) {
                        onChecked.accept(true);
                    }
                }
            );
        } else {
            onChecked.accept(true);
        }
    }

    public static void checkApplicationSpoofing(Context context, SecurityChecker securityChecker, Consumer<Boolean> onChecked) {
        SecurityChecker.SecurityCheck result = securityChecker.checkAppSpoofing();
        if (result instanceof SecurityChecker.SecurityCheck.Warning) {
            SecurityChecker.SecurityCheck.Warning warning = (SecurityChecker.SecurityCheck.Warning) result;
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(
                AppActivity.getContext(), 
                warning.message, 
                false,
                userAcknowledged -> {
                    if (userAcknowledged) {
                        onChecked.accept(true);
                    }
                }
            );
        } else {
            onChecked.accept(true);
        }
    }

    public static void checkKeyLoggerDetection(Context context, SecurityChecker securityChecker, Consumer<Boolean> onChecked) {
        SecurityChecker.SecurityCheck result = securityChecker.checkKeyLoggerDetection();
        if (result instanceof SecurityChecker.SecurityCheck.Warning) {
            SecurityChecker.SecurityCheck.Warning warning = (SecurityChecker.SecurityCheck.Warning) result;
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(
                AppActivity.getContext(), 
                warning.message, 
                false,
                userAcknowledged -> {
                    if (userAcknowledged) {
                        onChecked.accept(true);
                    }
                }
            );
        } else {
            onChecked.accept(true);
        }
    }

    public static void appSignatureCheck(Context context, SecurityChecker securityChecker, Consumer<Boolean> onChecked) {
        SecurityChecker.SecurityCheck result = securityChecker.appSignatureCompare();
        if (result instanceof SecurityChecker.SecurityCheck.Warning) {
            SecurityChecker.SecurityCheck.Warning warning = (SecurityChecker.SecurityCheck.Warning) result;
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(
                AppActivity.getContext(), 
                warning.message, 
                false,
                userAcknowledged -> {
                    if (userAcknowledged) {
                        onChecked.accept(true);
                    }
                }
            );
        } else {
            onChecked.accept(true);
        }
    }

    public static void checkInvoiceCall(Context context, SecurityChecker securityChecker, boolean inCall, Consumer<Boolean> onChecked) {
        if (inCall) {
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(
                AppActivity.getContext(),
                context.getString(R.string.ongoing_call_warning),
                false,
                userAcknowledged -> {
                    if (userAcknowledged) {
                        onChecked.accept(true);
                    }
                }
            );
        } else {
            onChecked.accept(true);
        }
    }

    public static void checkNetwork(Context context, SecurityChecker securityChecker, Consumer<Boolean> onChecked) {
        SecurityChecker.SecurityCheck result = securityChecker.checkNetworkSecurity();
        if (result instanceof SecurityChecker.SecurityCheck.Warning) {
            SecurityChecker.SecurityCheck.Warning warning = (SecurityChecker.SecurityCheck.Warning) result;
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(
                AppActivity.getContext(), 
                warning.message, 
                false,
                userAcknowledged -> {
                    if (userAcknowledged) {
                        onChecked.accept(true);
                    }
                }
            );
        } else {
            onChecked.accept(true);
        }
    }

    public static void showSecurityDialogForCheck(Context context, SecurityChecker.SecurityCheck checkResult, boolean isCritical, Consumer<Boolean> onResponse) {
        if (checkResult instanceof SecurityChecker.SecurityCheck.Critical) {
            SecurityChecker.SecurityCheck.Critical critical = (SecurityChecker.SecurityCheck.Critical) checkResult;
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(
                context,
                critical.message,
                true,
                null
            );
        } else if (checkResult instanceof SecurityChecker.SecurityCheck.Warning) {
            SecurityChecker.SecurityCheck.Warning warning = (SecurityChecker.SecurityCheck.Warning) checkResult;
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(
                context,
                warning.message,
                false,
                onResponse
            );
        }
    }
}

File: protect/src/main/java/com/webileapps/safeguard/CyberUtils.kt
package com.webileapps.safeguard

import android.content.Context
import android.content.Intent
import android.content.pm.PackageManager
import android.net.ConnectivityManager
import android.net.NetworkCapabilities
import android.os.Build
import android.provider.Settings
import android.widget.Toast

fun Context.showToast(message: String) {
    Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
}

fun Context.checkRoot(securityChecker: SecurityChecker, onChecked: (Boolean) -> Unit) {
    val result = securityChecker.checkRootStatus()
    when (result) {
        is SecurityChecker.SecurityCheck.Critical -> {
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(AppActivity.context, result.message, isCritical = true)
            onChecked(false)
        }
        is SecurityChecker.SecurityCheck.Warning -> {
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(AppActivity.context, result.message, isCritical = false) { userAcknowledged ->
                if (userAcknowledged) {
                    onChecked(true)
                }
            }
        }
        else -> {
            onChecked(true)
        }
    }
}

fun Context.checkDeveloperOptions(securityChecker: SecurityChecker, onChecked: (Boolean) -> Unit) {
    when (val result = securityChecker.checkDeveloperOptions()) {
        is SecurityChecker.SecurityCheck.Critical -> {
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(AppActivity.context, result.message, true)
            onChecked(false)
        }
        is SecurityChecker.SecurityCheck.Warning -> {
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(AppActivity.context, result.message, false) { userAcknowledged ->
                if (userAcknowledged) {
                    onChecked(true)
                }
            }
        }
        else -> onChecked(true)
    }
}

fun Context.checkMalware(securityChecker: SecurityChecker, onChecked: (Boolean) -> Unit) {
    when (val result = securityChecker.checkMalwareAndTampering()) {
        is SecurityChecker.SecurityCheck.Critical -> {
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(AppActivity.context, result.message, true)
        }
        is SecurityChecker.SecurityCheck.Warning -> {
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(AppActivity.context, result.message, false) { userAcknowledged ->
                if (userAcknowledged) {
                    onChecked(true)
                }
            }
        }
        else -> onChecked(true)
    }
}

fun Context.checkScreenMirroring(securityChecker: SecurityChecker, onChecked: (Boolean) -> Unit) {
    when (val result = securityChecker.checkScreenMirroring()) {
        is SecurityChecker.SecurityCheck.Warning -> {
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(AppActivity.context, result.message, false) { userAcknowledged ->
                if (userAcknowledged) {
                    onChecked(true)
                }
            }
        }
        else -> onChecked(true)
    }
}

fun Context.checkApplicationSpoofing(securityChecker: SecurityChecker, onChecked: (Boolean) -> Unit) {
    when (val result = securityChecker.checkAppSpoofing()) {
        is SecurityChecker.SecurityCheck.Warning -> {
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(AppActivity.context, result.message, false) { userAcknowledged ->
                if (userAcknowledged) {
                    onChecked(true)
                }
            }
        }
        else -> onChecked(true)
    }
}

fun Context.checkKeyLoggerDetection(securityChecker: SecurityChecker, onChecked: (Boolean) -> Unit) {
    when (val result = securityChecker.checkKeyLoggerDetection()) {
        is SecurityChecker.SecurityCheck.Warning -> {
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(AppActivity.context, result.message, false) { userAcknowledged ->
                if (userAcknowledged) {
                    onChecked(true)
                }
            }
        }
        else -> onChecked(true)
    }
}

fun Context.checkNetwork(securityChecker: SecurityChecker, onChecked: (Boolean) -> Unit) {
    when (val result = securityChecker.checkNetworkSecurity()) {
        is SecurityChecker.SecurityCheck.Warning -> {
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(AppActivity.context, result.message, false) { userAcknowledged ->
                if (userAcknowledged) {
                    onChecked(true)
                }
            }
        }
        else -> onChecked(true)
    }
}

fun Context.showSecurityDialogForCheck(checkResult: SecurityChecker.SecurityCheck, isCritical: Boolean, onResponse: ((Boolean) -> Unit)? = null) {
    when (checkResult) {
        is SecurityChecker.SecurityCheck.Critical -> {
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(AppActivity.context, checkResult.message, isCritical = true, onResponse)
        }
        is SecurityChecker.SecurityCheck.Warning -> {
            SecurityConfigManager.getSecurityChecker().showSecurityDialog(AppActivity.context, checkResult.message, isCritical = false, onResponse)
        }
        else -> {}
    }
}